### PR TITLE
Document IIS instrumentation

### LINKED ISSUE #632: Document how to instrument an ASP.NET application deployed on IIS
Per #631

### LINKED ISSUE #689: Do not require installing `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule` when instrumenting an ASP.NET application
1. Describe steps to instrument ASP.NET (.NET Framework) application without installing  `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule`
1. Update the test-application to not reference `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule` (if possible)

_Originally posted by @pellared in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/688#discussion_r876992545_

I found it as the easiest way to add `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule` ASP.NET HTTP module which is required by [OpenTelemetry.Instrumentation.AspNet](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.AspNet). This is also how our `test-application` was created.

I think it should be possible to simply edit `Web.config` to add the `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule` as the required assemblies are under `$trace-home/net462`. I will check it tomorrow.

I would prefer to make baby steps. Therefore I am creating a GH issue for it. Please resolve if you are fine with such an approach.

### PR DESCRIPTION
**Pull request by pellared**
Fixes #632
Fixes #689

Changes proposed in this pull request:
- Document how to instrument an ASP.NET and ASP.NET Core application deployed on IIS 
- Remove necessary references from `TestApplication.AspNet`

Docs preview: https://github.com/pellared/opentelemetry-dotnet-instrumentation/blob/docs-iis/docs/iis-instrumentation.md

## Testing

I was testing against an ASP.NET Web API application created using the Visual Studio template.

```
otel-collector_1  | 2022-05-19T11:42:39.726Z    DEBUG   loggingexporter/logging_exporter.go:49  ResourceSpans #0
otel-collector_1  | Resource SchemaURL:
otel-collector_1  | Resource labels:
otel-collector_1  |      -> service.name: STRING(unknown_service:w3wp)
otel-collector_1  | ScopeSpans #0
otel-collector_1  | ScopeSpans SchemaURL:
otel-collector_1  | InstrumentationScope OpenTelemetry.Instrumentation.AspNet.Telemetry 1.0.0.0
otel-collector_1  | Span #0
otel-collector_1  |     Trace ID       : 8a066413b0daa361ddb238e95d967ce2
otel-collector_1  |     Parent ID      :
otel-collector_1  |     ID             : 8a06ce6e7f8884b7
otel-collector_1  |     Name           : api/{controller}/{id}
otel-collector_1  |     Kind           : SPAN_KIND_SERVER
otel-collector_1  |     Start time     : 2022-05-19 11:42:36.612067 +0000 UTC
otel-collector_1  |     End time       : 2022-05-19 11:42:36.6126715 +0000 UTC
otel-collector_1  |     Status code    : STATUS_CODE_UNSET
otel-collector_1  |     Status message :
otel-collector_1  | Attributes:
otel-collector_1  |      -> http.host: STRING(localhost:8881)
otel-collector_1  |      -> http.method: STRING(GET)
otel-collector_1  |      -> http.target: STRING(/api/values)
otel-collector_1  |      -> http.user_agent: STRING(Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36)
otel-collector_1  |      -> http.url: STRING(http://localhost:8881/api/values)
otel-collector_1  |      -> http.status_code: INT(200)
otel-collector_1  |      -> http.route: STRING(api/{controller}/{id})
````

I was testing it also against an ASP.NET Core Web API application created using the Visual Studio template.

```
2022-05-20T10:18:30.736Z DEBUG loggingexporter/logging_exporter.go:49 ResourceSpans #0
Resource SchemaURL: 
Resource labels:
     -> service.name: STRING(CoreWebApp1)
ScopeSpans #0
ScopeSpans SchemaURL: 
InstrumentationScope OpenTelemetry.Instrumentation.AspNetCore 1.0.0.0
Span #0
    Trace ID       : ea73bcc63f83d87d6e15a264cd8b1330
    Parent ID      : 
    ID             : ab98e96dea4228c9
    Name           : WeatherForecast
    Kind           : SPAN_KIND_SERVER
    Start time     : 2022-05-20 10:18:26.1864498 +0000 UTC
    End time       : 2022-05-20 10:18:26.5124016 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message : 
Attributes:
     -> http.host: STRING(localhost:8885)
     -> http.method: STRING(GET)
     -> http.target: STRING(/WeatherForecast)
     -> http.url: STRING(http://localhost:8885/WeatherForecast)
     -> http.user_agent: STRING(Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36)
     -> http.route: STRING(WeatherForecast)
     -> http.status_code: INT(200)
```

### CODE REVIEW

File: docs/iis-instrumentation.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+You can instrumental all ASP.NET 4.x application deployed to IIS
+by setting the required environment variables for
+`W3SVC` and `WAS` Windows Services as described in [windows-service-instrumentation.md](windows-service-instrumentation.md).
+
+> Unfortunately it is not possible to set distinct environment variables
+  values for the instrumented ASP.NET application.
+  Therefore all applications will share
+  the same configuration (e.g. `OTEL_SERVICE_NAME`).
+
+Install the [`OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule`](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/)
```
[the remark this answers is not in the document]
Comment (pellared) [reply docs/iis-instrumentation.md:14]: I found it as the easiest way to add `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule` ASP.NET HTTP module which is required by [OpenTelemetry.Instrumentation.AspNet](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.AspNet). This is also how our `test-application` was created.

I think it should be possible to simply edit `Web.config` to add the `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule` as the required assemblies are under `$trace-home/net462`. I will check it tomorrow.

I would prefer to make baby steps. Therefore I am creating a GH issue for it. Please resolve if you are fine with such an approach.

[the remark this answers is not in the document]
Comment (pellared) [reply docs/iis-instrumentation.md:14]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/689

File: docs/iis-instrumentation.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Instrument an ASP.NET application deployed on IIS
+
+## Instrument an ASP.NET 4.x application
+
+You can instrumental all ASP.NET 4.x application deployed to IIS
+by setting the required environment variables for
+`W3SVC` and `WAS` Windows Services as described in [windows-service-instrumentation.md](windows-service-instrumentation.md).
+
+> Unfortunately it is not possible to set distinct environment variables
```
Comment (rajkumar-rangaraj) [inline docs/iis-instrumentation.md:9]: There is an option to set environment variables at the application pool level. They could use this feature as a workaround for ASP.NET app.

Ref: https://docs.microsoft.com/en-us/iis/configuration/system.applicationhost/applicationpools/add/environmentvariables/

File: docs/iis-instrumentation.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rajkumar-rangaraj) [reply docs/iis-instrumentation.md:14]: There is another option to extract the content of `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule` package and add the module entry with file location either in applicationHost.config or web.config.

[the remark this answers is not in the document]
Comment (pellared) [reply docs/iis-instrumentation.md]: I update the docs and test application to not reference `OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule`.

File: docs/iis-instrumentation.md (hunk 2 above)
Comment (pellared) [reply docs/iis-instrumentation.md]: Thanks. That is great.

**Review by Kielek — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Review by nrcventura — APPROVED** [review]

**Pull request merged**